peachjam/views/documents.py

```diff
@@ -9,6 +11,98 @@
 from peachjam.registry import registry
 
 
+class RedirectResolver:
+
+    RESOLVER_MAPPINGS = {
+        "africanlii": {
+            "country_code": "aa",
+            "domain": "agp.africanlii.org",
+        },
+        "ghalii": {
+            "country_code": "gh",
+            "domain": "new.ghalii.org",
+        },
+        "lawlibrary": {
+            "country_code": "za",
+            "domain": "lawlibrary.org.za",
+        },
+        "lesotholii": {
+            "country_code": "ls",
+            "domain": "new.lesotholii.org",
+        },
+        "malawilii": {
+            "country_code": "mw",
+            "domain": "malawilii.org",
+        },
+        "namiblii": {
+            "country_code": "na",
+            "domain": "namiblii.org",
+        },
+        "senlii": {
+            "country_code": "sn",
+            "domain": "senlii.org",
+        },
+        "seylii": {
+            "country_code": "sc",
+            "domain": "new.seylii.org",
+        },
+        "sierralii": {
+            "country_code": "sl",
+            "domain": "new.sierralii.org",
+        },
+        "tanzlii": {
+            "country_code": "tz",
+            "domain": "new.tanzlii.org",
+        },
+        "tcilii": {
+            "country_code": "tc",
+            "domain": "new.tcilii.org",
+        },
+        "ulii": {
+            "country_code": "ug",
+            "domain": "new.ulii.org",
+        },
+        "zambialii": {
+            "country_code": "zm",
+            "domain": "new.zambialii.org",
+        },
+        "zanzibarlii": {
+            "place_code": "tz-znz",
+            "domain": "zanzibarlii.org",
+        },
+        "zimlii": {
+            "country_code": "zw",
+            "domain": "zimlii.org",
+        },
+    }
+
+    def __init__(self, app_name):
+        self.current_authority = self.RESOLVER_MAPPINGS[app_name.lower()]
+
+    def get_url_for_frbr_uri(self, parsed_uri):
+        best_authority = self.get_best_authority(parsed_uri)
+        if best_authority != self.current_authority["domain"]:
+            return best_authority
+        return None
+
+    def get_best_authority(self, parsed_uri):
+        country_code = parsed_uri.country
+        place_code = parsed_uri.place
+
+        domain = None
+        if country_code != place_code:
+            for key, mapping in self.RESOLVER_MAPPINGS.items():
+                if mapping.get("place_code") == place_code:
+                    domain = mapping.get("domain")
```

Comment:
I think you can simply `return mapping.get("domain")` here (or `return mapping`), no need to keep looping.

Ditto for line 102 below

Reply:
right, changed